packages/aggregator/src/analytics/aggregateAnalytics/aggregations/utils/getContinuousPeriodsForAnalytics.js

```diff
@@ -21,13 +21,14 @@ export const getContinuousPeriodsForAnalytics = (
   const periodsInAnalytics = analytics.map(analytic =>
     convertToPeriod(analytic.period, aggregationPeriod),
   );
+  const sortedPeriodsInAnalytics = periodsInAnalytics.sort(compareAsc);
   const endPeriod =
     !continueTilCurrentPeriod && periodsInAnalytics.length
-      ? Math.max(...periodsInAnalytics).toString()
+      ? sortedPeriodsInAnalytics[sortedPeriodsInAnalytics.length - 1].toString() // Max
       : getCurrentPeriod(aggregationPeriod);
```

Comment:
For weekly period we have string items in periodsInAnalytics (e.g. ['2021W22', '2021W19', '2021W25', ...]), which couldn't play with Math.max() and Math.min()